src/Controller/PublishersController.php

```diff
@@ -104,4 +104,21 @@ public function edit(Request $request, int $id): Response
             'publisher_form' => $form->createView()
         ]);
     }
+
+    /**
+     * @Route ("/deletePublisher/{id}", name="app_deletePublisher")
+     */
+    public function delete(Publisher $publisher): Response
+    {
+        if ($publisher->getGames()->isEmpty()) {
+            $response = $this->dataFactory->deletePublisher($publisher);
+            $type = "success";
+        } else {
+            $response = "Delete publisher games first.";
+            $type = "warning";
+        }
+        $this->addFlash($type, $response);
```

Comment:
Try to leave an empty line after an if or else statement, not a must - but it's usually the case when it comes to accepted coding standards. (unless the method ends with that statement of course).